src/base/basic.lisp

```diff
@@ -257,11 +257,21 @@ The thrid argument PROP is a property to remove."
                      (count-characters (buffer-start-point buffer)
                                        (buffer-end-point buffer))))
 
+(defgeneric global-mode-region-beginning (global-mode &optional buffer))
```

Comment:
The concept of global-mode does not exist at the time of loading lem-base.
This causes interdependencies between lem-base and lem-core.

Reply:
That's why I'm using keywords to dispatch the global modes, as in this stage lem/base doesn't have the concept yet. But we can always work with keywords.